internal/pkg/consul/client_test.go

```diff
@@ -133,6 +133,63 @@ func TestRegisterWithPingCallback(t *testing.T) {
 	assert.True(t, receivedPing, "Never received health check ping")
 }
 
+func TestRegisterCustomWithPingCallback(t *testing.T) {
+	doneChan := make(chan bool)
+	receivedPing := false
+
+	route := "/test/route"
+
+	// Setup a server to simulate the service for the health check callback
+	server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		if strings.Contains(request.URL.Path, route) {
+
+			switch request.Method {
+			case "GET":
+				receivedPing = true
+
+				writer.Header().Set("Content-Type", "text/plain")
+				_, _ = writer.Write([]byte("pong"))
+
+				doneChan <- true
+			}
+		}
+	}))
+	defer server.Close()
+
+	// Figure out which port the simulated service is running on.
+	serverUrl, _ := url.Parse(server.URL)
+	serverPort, _ := strconv.Atoi(serverUrl.Port())
+
+	client := makeConsulClient(t, getUniqueServiceName(), serverPort, true)
+	// Make sure service is not already registered.
+	_ = client.consulClient.Agent().ServiceDeregister(client.serviceKey)
+	_ = client.consulClient.Agent().CheckDeregister(client.serviceKey)
+
+	// Try to clean-up after test
+	defer func(client *consulClient) {
+		_ = client.consulClient.Agent().ServiceDeregister(client.serviceKey)
+		_ = client.consulClient.Agent().CheckDeregister(client.serviceKey)
+	}(client)
+
+	id := "check-id"
+	name := "check-name"
+
+	// Register the service endpoint and health check callback
+	err := client.RegisterCheck(id, name, "", route, "5s")
+
+	if !assert.NoError(t, err) {
+		t.Fatal()
+	}
```

Comment:
use require.NoError(t,err) here and all other similar checks.
It will call Fatal is err not nil

Reply:
Just pushed.  I'm always shy about making changes like that but it does look way better.  I kind of prefer require in general it works more like other assertion libraries I've used in the past.

Reply:
I just did a find/replace on assert to require btw.  I only saw a couple assertions that weren't using the pattern you pointed out and they all seemed to be the last line in the test.

Reply:
Thanks!